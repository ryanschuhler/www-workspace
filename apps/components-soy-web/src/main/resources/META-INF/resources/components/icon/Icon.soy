{namespace Icon}

/**
 * Icon
 */
{template .render}
  {@param isMaterialIcon: bool}
  {@param symbol: string}
  {@param? cssClass: string}
  {@param? pathThemeImages: string}

  {let $classes kind="text"}
    icon-root
    {sp}pario-icon
    {sp}lexicon-icon

    {if $isMaterialIcon}
      {sp}material-icons
    {/if}

    {if $cssClass}
      {sp}{$cssClass}
    {/if}
  {/let}

  {if $isMaterialIcon}
    <i class="{$classes}">{$symbol}</i>
  {else}
    <svg class="{$classes}">
      <use xlink:href="{$pathThemeImages}/pario/icons.svg#{$symbol}"></use>
    </svg>
  {/if}
{/template}
