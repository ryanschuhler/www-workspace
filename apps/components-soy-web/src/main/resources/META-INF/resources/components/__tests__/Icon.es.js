'use strict';

import Icon from '../icon/Icon.es';

let component;

describe('Icon', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {};
  });

  it('should render material-icon', () => {
    component = new Icon({
      isMaterialIcon: true,
      symbol: 'add'
    });

    expect(component).toMatchSnapshot();
  });
});
