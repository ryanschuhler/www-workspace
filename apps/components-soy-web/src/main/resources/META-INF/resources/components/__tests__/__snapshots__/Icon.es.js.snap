// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render material-icon 1`] = `<i class="icon-root pario-icon lexicon-icon material-icons">add</i>`;
